internal/metrics/listener.go

```diff
@@ -61,9 +63,18 @@ func MakeListener(config Config) Listener {
 		apiClient: apiClient,
 		config:    &config,
 		processor: nil,
+		trace:     config.Trace,
 	}
 }
 
+func (l *Listener) Trace() bool {
+	return l.trace
+}
+
+func (l *Listener) Name() string {
+	return "Datadog-Metrics"
```

Comment:
I don't have a super strong opinion, but maybe either get rid of the hyphen here or convert it all to lowercase?